core/src/main/java/com/netflix/iceberg/util/ByteBuffers.java

```diff
@@ -0,0 +1,38 @@
+package com.netflix.iceberg.util;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+
+public class ByteBuffers {
+
+  public static ByteBuffer copy(ByteBuffer toCopy) {
+    return ByteBuffer.wrap(toByteArray(toCopy));
+  }
+
+  public static byte[] toByteArray(ByteBuffer buffer) {
```

Comment:
This is taken from `SerializableByteBufferMap`. One implementation detail that we should review throughout this patch is how to avoid copying bytes as much as possible. I over-compensate for correctness by defensive copying these byte arrays frequently. We might be able to get away with something more efficient in some places.